test/WebSites/DispatchingWebSite/Startup.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matchers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace DispatchingWebSite
+{
+    public class Startup
+    {
+        private static readonly byte[] _homePayload = Encoding.UTF8.GetBytes("Dispatcher sample endpoints:" + Environment.NewLine + "/plaintext");
+
+        // Set up application services
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDispatcher(options =>
+            {
+                options.DataSources.Add(new DefaultEndpointDataSource(new[]
```

Comment:
You can also directly register a data source in DI. Ordering of datasources isn't super meaningful so we provide that as a convenience.